stdlib/stdlib_docs.rs

```diff
@@ -0,0 +1,204 @@
+use crate::ModuleMap;
+
+use std::collections::BTreeMap;
+use std::fs;
+use std::io::Write;
+use std::path::Path;
+
+// CONSTANTS
+// ================================================================================================
+
+pub const COMMENT_PREFIX: &str = "#";
+pub const FUNC_PREFIX: &str = "export.";
+pub const MODULE_COMMENT_PREFIX: &str = "#!";
+
+/// Holds a stdlib function name and comments for docs purposes.
+pub struct Function {
+    name: String,
+    comments: Vec<String>,
+}
+
+/// Holds a stdlib module that contains a list of functions and comments.
+pub struct Module {
+    name: String,
+    section: String,
+    functions: Vec<Function>,
+    comments: Vec<String>,
+}
+
+impl Module {
+    pub fn new(ns: String) -> Self {
+        let parts: Vec<&str> = ns.split("::").collect();
+        let module_name = parts[parts.len() - 1];
+        let section_name = parts[parts.len() - 2];
+        Module {
+            name: String::from(module_name),
+            section: String::from(section_name),
+            functions: Vec::new(),
+            comments: Vec::new(),
+        }
+    }
+
+    pub fn markdown_file_name(&self) -> String {
+        format!("{}_{}.md", self.name, self.section)
+    }
+}
+
+/// Holds the stdlib modules and corresponding functions.
+pub struct Stdlib {
+    modules: BTreeMap<String, Module>,
+}
+
+impl Stdlib {
+    pub fn new() -> Self {
+        Stdlib {
+            modules: BTreeMap::new(),
+        }
+    }
+}
+
+/// A renderer renders a stdlib struct into a particular doc format and index (e.g: markdown, etc)
+trait Renderer {
+    // Render writes out the document files into the output directory
+    fn render(stdlib: &Stdlib, output_dir: &str);
+}
+
+struct MarkdownRenderer {}
+
+impl Renderer for MarkdownRenderer {
+    fn render(stdlib: &Stdlib, output_dir: &str) {
+        // Write per module markdown file
+        for (ns, module) in &stdlib.modules {
+            let file_name = module.markdown_file_name();
+            let file_path = Path::new(output_dir).join(file_name);
+            let mut f = fs::OpenOptions::new()
+                .write(true)
+                .append(true)
+                .create(true)
+                .open(file_path)
+                .expect("unable to open stdlib markdown file");
+
+            // Render modules into markdown
+            f.write_all(module.comments.join("\n").as_bytes())
+                .expect("unable to write module comments");
+            let header = format!(
+                "\n## {}\n| Instruction | Description | \n| ----------- | ------------- |\n",
```

Comment:
We should say "Procedure" instead of "Instruction".

Reply:
fixed